src/components/Navigation/components/Item/Item.tsx

```diff
@@ -87,13 +88,27 @@ export function Item({
   const {isNavigationCollapsed} = useMediaQuery();
   const {location, onNavigationDismiss} = useContext(NavigationContext);
   const [expanded, setExpanded] = useState(false);
+  const [keyFocused, setKeyFocused] = useState(false);
 
   useEffect(() => {
     if (!isNavigationCollapsed && expanded) {
       setExpanded(false);
     }
   }, [expanded, isNavigationCollapsed]);
 
+  const handleKeyUp = useCallback(
```

Comment:
A way to test this is to count how many times this is called with and without a new value for `keyFocused` to get an idea of how effective this is.